packages/instantsearch.js/src/connectors/related-products/connectRelatedProducts.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  createDocumentationMessageGenerator,
+  checkRendering,
+  noop,
+} from '../../lib/utils';
+
+import type { Connector, TransformItems, Hit, BaseHit } from '../../types';
+import type {
+  RecommendResultItem,
+  SearchParameters,
+} from 'algoliasearch-helper';
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'related-products',
+  connector: true,
+});
+
+export type RelatedProductsRenderState<THit extends BaseHit = BaseHit> = {
+  /**
+   * The matched recommendations from the Algolia API.
+   */
+  recommendations: Array<Hit<THit>>;
+};
+
+export type RelatedProductsConnectorParams<THit extends BaseHit = BaseHit> = {
+  /**
+   * The `objectID` of the item to get related products from.
+   */
+  objectID: string;
```

Comment:
The underlying implementation of multi object ids will be handled next sprint, but we can already switch to `objectIDs` in the connector.